app/models/pglogical_subscription.rb

```diff
@@ -1,36 +1,75 @@
 require 'pg/dsn_parser'
 require 'pg/logical_replication'
+require 'query_relation'
 
-class PglogicalSubscription < ActsAsArModel
-  set_columns_hash(
-    :id                   => :string,
-    :status               => :string,
-    :dbname               => :string,
-    :host                 => :string,
-    :user                 => :string,
-    :password             => :string,
-    :port                 => :integer,
-    :provider_region      => :integer,
-    :provider_region_name => :string
-  )
-
-  def self.find(*args)
-    case args.first
-    when :all then find_all
-    when :first, :last then find_one(args.first)
-    else find_id(args.first)
+class PglogicalSubscription
+  include Vmdb::Logging
+  include ActiveModel::Model
+  extend QueryRelation::Queryable
+
+  attr_accessor :id, :status, :dbname, :host, :user, :password, :port, :provider_region, :provider_region_name
+
+  # A list of attributes typically used on inspection, and necessary
+  # for ActiveModel::AttributeMethods, if used.
+  #
+  def attributes
+    {
+      :id                   => @id,
+      :status               => @status,
+      :dbname               => @dbname,
+      :host                 => @host,
+      :user                 => @user,
+      :password             => @password,
+      :port                 => @port,
+      :provider_region      => @provider_region,
+      :provider_region_name => @provider_region_name
+    }
+  end
+
+  def self.connection
+    ActiveRecord::Base.connection
+  end
+
+  def connection
+    self.class.connection
+  end
+
+  # Interface method required by QueryRelation. If +mode+ is not a hash option,
+  # it assumes the argument is an id.
+  #
+  def self.search(mode, options = {})
+    collection = subscriptions.collect { |s| new(subscription_to_columns(s)) }
+    collection = filter_collection(collection, options) if options.present?
+
+    case mode
+    when :all
+      collection
+    when :first
+      collection.first
+    when :last
+      collection.last
+    else
+      collection.find { |e| e.id == mode } or raise ActiveRecord::RecordNotFound
```

Comment:
Prefer `||` over `or`

Reply:
Rubocop is wrong, that's not the logic we want. It's not a boolean expression. I'm surprised rubocop gets this wrong.

Reply:
https://github.com/rubocop-hq/ruby-style-guide#no-and-or-or

I don't agree, this is an age-old idiom.

Reply:
I think consistency with the rest of the code base is more important (I tried to find another usage of `or` and failed). If you really don't want to use the logical operator here you could do something like this, but I still think using `||` would be preferable as, like I said, that's what we do everywhere else.

```ruby
collection.find { |e| e.id == mode }.tap { |result| raise ActiveRecord::RecordNotFound unless result }
```

Reply:
Ok, updated.